llama-index-integrations/vector_stores/llama-index-vector-stores-couchbase/llama_index/vector_stores/couchbase/base.py

```diff
@@ -0,0 +1,457 @@
+"""
+Couchbase Vector store interface.
+"""
+
+from typing import Any, Dict, List, Optional
+
+import logging
+
+from llama_index.core.schema import BaseNode, MetadataMode, TextNode
+from llama_index.core.vector_stores.types import (
+    BasePydanticVectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+    MetadataFilters,
+)
+from llama_index.core.vector_stores.utils import (
+    node_to_metadata_dict,
+    metadata_dict_to_node,
+)
+from pydantic import PrivateAttr
+
+logger = logging.getLogger(__name__)
+
+
+def _to_couchbase_filter(standard_filters: MetadataFilters) -> Dict[str, Any]:
+    """
+    Convert standard filters to Couchbase filter.
+
+    Args:
+        standard_filters (str): Standard Llama-index filters.
+
+    Returns:
+        Dictionary with Couchbase search filters.
+    """
+    if len(standard_filters.legacy_filters()) == 1:
```

Comment:
I'm not sure if couchebase supports it, but `legacy_filters` means only `==` filters get used

In reality, we support a ton of filter arguments, and also `and` and `or` operators. Check out an integration like pinecone to see it in action. If couchbase supports this, would be nice to support it here too

Reply:
Couchbase supports filters as well but they are not the same filters as LlamaIndex.  For example, `>=` would have to be converted to a range query. 
So, I put in only the one filter that is the same. It is similar to Elastic. 
However, users can put in the couchbase specific filters via `cb_search_options`

Reply:
I have found a way to support some more metadatafilters. I have added it in 2dc715400fd713a0dfc423d811647e62e3bc1bdc

Reply:
Awesome, thanks!